### DISCUSSION TITLE
Mc John Modular Rig

### DISCUSSION THREAD

**Discussion by McJohnSrl** [JK-Rolling/DuinoI2C_ESP#6] [Show and tell]
I designed a modular miner Rig and took inspiration from JK-Rolling's design.
The idea of ​​making a passive bus comes from the S100 bus standard used in computers with CP/M in the 70s/80s.
The Bus is very simple, it carries the +5V power supply and GND ground and the I2C signals.
Each slot of the Bus has its own filter and Bypass capacitors, this complicates the scheme a little but makes the miner very stable in its continuous operation.
The Bus can be reused for any Master and Slave configuration, in the photo a master module with ESP12F (aka ESP8266EX) and ATMEGA328 slave modules is used.
This mining rig produces 120 Duco/day.
Previously I have already made a master module with ESP32 and now I am designing a new slave module with ATTiny 85.

![Minatore_02_800](https://github.com/user-attachments/assets/ad0b35ee-4eeb-4b28-b851-85c78e96ce2d)

![JK-Rolling_Avr](https://github.com/user-attachments/assets/a554b7aa-fc71-4c6f-828a-e980b7819aff)

If anyone likes this project, you can order it [here](https://shop.mcjohn.it/en/miners/152-duino-coin-miner-20-avr-modules.html).

Thanks :)

**Comment by McJohnSrl** [comment]
![Pcb_bus](https://github.com/user-attachments/assets/6b34bf50-f710-423b-b306-631f05cfdc80)

![Pcb_Esp12f](https://github.com/user-attachments/assets/983cb6ce-41f9-405e-b953-6f44a159c57e)

![Pcb_Avr](https://github.com/user-attachments/assets/f7bc5e78-5729-464d-9bf8-c20f12e670b9)

![Pcb_Display](https://github.com/user-attachments/assets/be95dddb-9699-420b-9d21-45998f2c2341)

**Comment by JK-Rolling** [comment]
I can see everything from ESP, OLED to AVR are custom made. good job!
Everyone, if you'd like to get your hands on this awesome mining rig, you may get them from [https://shop.mcjohn.it/]( https://shop.mcjohn.it/en/miners/152-duino-coin-miner-20-avr-modules.html)

**Reply by McJohnSrl** [reply]
This is my next slave module with Attiny 85.

![immagine](https://github.com/user-attachments/assets/72107aba-8299-4b6d-86d3-969905fd3d4a)

![immagine](https://github.com/user-attachments/assets/4bbe80df-d99c-41c1-b31c-65f3e82d2f18)

**Comment by McJohnSrl** [comment]
All parts of the Miner Rig can also be ordered individually. 
https://shop.mcjohn.it/en/28-miners

**Comment by McJohnSrl** [comment]
Previously I also designed a module with ESP32 which can be used as a master module or slave module, it depends on the firmware loaded.

![immagine](https://github.com/user-attachments/assets/fcd98619-2d8d-470f-ace9-1141a2269f12)

![immagine](https://github.com/user-attachments/assets/17ec543e-08f0-4d90-8e33-a239c02d6e25)

**Comment by McJohnSrl** [comment]
If this Miner Rig will be successful and there will be user requests, I might also design a module with RP3040 (Raspberry pico).

**Comment by xadosss** [comment]
Hello, i am doing a board similar to yours, and i was wondering if you could eventualy review it be cause its my first doing a custom pcb and i dont know if i connected everything correctly.
